README.md

```diff
@@ -217,153 +217,194 @@ varsig-body = *OCTET; Zero or more segments required by the kind of varsig (e.g.
 
 For example, here is an EdDSA signature for some content encoded as DAG-PB:
 
-`0x34ed01ae3784f03f9ee1163382fa6efa73b0c31ecf58c899c836709303ba4621d1e6df20e09aaa568914290b7ea124f5b38e70b9b69c7de0d216880eac885edd41c302`
+`%x34ed01ae3784f03f9ee1163382fa6efa73b0c31ecf58c899c836709303ba4621d1e6df20e09aaa568914290b7ea124f5b38e70b9b69c7de0d216880eac885edd41c302`
 
 ### 3.1 Varsig Prefix
 
-The varsig prefix MUST be `0x34`.
+The varsig prefix MUST be `%x34`.
 
 ### 3.2 Signature Header
 
 The prefix of the signature algorithm. This is often the [multicodec] of the associated public key, but MAY be unique for the signature type. The code MAY live outside the multicodec table. This field MUST act as a discriminant for how many expected fields come in the varsig body, and what each of them mean.
 
 ### 3.3 Varsig Body
 
-The varsig body MUST consist of zero or more segments, and MUST be defined by the signature algorithm.
+The varsig body MUST consist of one or more segments, and MUST be defined by the signature algorithm.
 
 Some examples include:
 
 * Raw signature bytes only
 * CID of [DKIM] certification transparency record, and raw signature bytes
-* Hash algorithm multicodec prefix, signature counter, HMAC, and raw signature bytes
+* Hash algorithm multicodec prefix, data encoding prefix, signature counter, nonce, HMAC, and raw signature bytes
 
-# 4 Registry of Common Signature Algorithms
+# 4 Payload Encoding
+
+The [IPLD] data model is encoding agnostic by design. This is very convenient in many applications, such as making for very convenient conversions between types for transmission versus encoding. Unfortunately signatures require signing over specific bytes, and thus over a specific encoding of the data.
+
+To facilitate this, the type `encoding-info` MAY be used:
+
+``` abnf
+encoding-info
+  = %x55   ; Raw bytes multicodec prefix
+  / %x70   ; DAG-PB multicodec prefix
+  / %x71   ; DAG-CBOR multicodec prefix
+  / %x0129 ; DAG-JSON multicodec prefix
+  / %x6A77 ; JWT
+  / %xE191 eip-191-body encoding-info ; EIP-191
```

Comment:
Similarly, eip191 only works over a string. It would need to be paired with an IPLD codec, e.g. dag-cbor, to get the byte string to sign over first. For CACAO specifically we would want to define a new IPLD codec that takes a CACAO object and outputs an ascii SIWE string.

Reply:
This was my intention with the recursive `encoding-info`. For example:

```abnf
;    Ed25519     DAG-CBOR
;      |            |
;      v            v
%x34 (%xed (%xE191 %x71) <signature>)
;            ^
;            |
;         EIP-191
```

Does that make sense?

Reply:
> only works over a string

Also to clarify: you mean a bytestring, not an ASCII/UTF8 string, correct?

Reply:
Yes. Technically the `personal_sign` eth-rpc method accepts both utf8 and hex encoded byte strings. But ofc it's encoded as a bytestring before it gets prepended and hashed.